litestar/_openapi/parameters.py

```diff
@@ -75,181 +74,146 @@ def list(self) -> list[Parameter]:
         return list(self._parameters.values())
 
 
-def create_parameter(
-    field_definition: FieldDefinition,
-    parameter_name: str,
-    path_parameters: tuple[PathParameterDefinition, ...],
-    schema_creator: SchemaCreator,
-) -> Parameter:
-    """Create an OpenAPI Parameter instance."""
-
-    result: Schema | Reference | None = None
-    kwarg_definition = (
-        field_definition.kwarg_definition if isinstance(field_definition.kwarg_definition, ParameterKwarg) else None
-    )
-
-    if any(path_param.name == parameter_name for path_param in path_parameters):
-        param_in = ParamType.PATH
-        is_required = True
-        result = schema_creator.for_field_definition(field_definition)
-    elif kwarg_definition and kwarg_definition.header:
-        parameter_name = kwarg_definition.header
-        param_in = ParamType.HEADER
-        is_required = field_definition.is_required
-    elif kwarg_definition and kwarg_definition.cookie:
-        parameter_name = kwarg_definition.cookie
-        param_in = ParamType.COOKIE
-        is_required = field_definition.is_required
-    else:
-        is_required = field_definition.is_required
-        param_in = ParamType.QUERY
-        parameter_name = kwarg_definition.query if kwarg_definition and kwarg_definition.query else parameter_name
-
-    if not result:
-        result = schema_creator.for_field_definition(field_definition)
-
-    schema = result if isinstance(result, Schema) else schema_creator.schemas[result.value]
-
-    examples_list = kwarg_definition.examples or [] if kwarg_definition else []
-    examples = get_formatted_examples(field_definition, examples_list)
-
-    return Parameter(
-        description=schema.description,
-        name=parameter_name,
-        param_in=param_in,
-        required=is_required,
-        schema=result,
-        examples=examples or None,
-    )
-
-
-def get_recursive_handler_parameters(
-    field_name: str,
-    field_definition: FieldDefinition,
-    dependency_providers: dict[str, Provide],
-    route_handler: BaseRouteHandler,
-    path_parameters: tuple[PathParameterDefinition, ...],
-    schema_creator: SchemaCreator,
-) -> list[Parameter]:
-    """Create and return parameters for a handler.
-
-    If the provided field is not a dependency, a normal parameter is created and returned as a list, otherwise
-    `create_parameter_for_handler()` is called to generate parameters for the dependency.
-    """
+class ParameterFactory:
```

Comment:
I'm not a huge fan of this "class with a single method that's called immediately" pattern. From what I've seen so far, it's only ever created and then has its `create_parameters_for_handler` method called directly; Everything else is only ever called internally.

I guess the intent here was to encapsulate some state, but maybe this class should be private to the module then, and wrapped in a public `create_parameters_for_handler` function then, which sets up the class and calls its `create_parameters_for_handler` method?